cpu/atmega_common/periph/uart.c

```diff
@@ -74,94 +78,193 @@ static mega_uart_t *dev[] = {
 };
 
 /**
- * @brief   Allocate memory to store the callback functions.
+ * @brief   Allocate memory to store the settings.
  */
 static uart_isr_ctx_t isr_ctx[UART_NUMOF];
+static uint16_t uart_brr[UART_NUMOF];
 
-static void _update_brr(uart_t uart, uint16_t brr, bool double_speed)
+static inline uint8_t _is_asleep(uart_t uart)
 {
-    dev[uart]->BRR = brr;
-    if (double_speed) {
-        dev[uart]->CSRA |= (1 << U2X0);
+#if defined(CPU_ATMEGA328P)
+    (void)uart;
+    return ((PRR >> PRUSART0) & 0x1);
+#else
+
+    if (uart == 0) {
+        return ((PRR0 >> PRUSART0) & 0x1);
+    }
+
+#if defined(CPU_ATMEGA1284P)
+    else if (uart == 1) {
+        /* ATmega1284P uses PRR0 instead of PRR1 */
+        return ((PRR0 >> PRUSART1) & 0x1);
+    }
+#else
+    else if (uart == 1) {
+        return ((PRR1 >> PRUSART1) & 0x1);
     }
+#endif /* defined(CPU_ATMEGA1284P) */
+
+#if (UART_NUMOF > 2)
+    else if (uart == 2) {
+        return ((PRR1 >> PRUSART2) & 0x1);
+    }
+#endif
+
+#if (UART_NUMOF > 3)
+    else if (uart == 3) {
+        return ((PRR1 >> PRUSART3) & 0x1);
+    }
+#endif
+
+#endif /* defined(CPU_ATMEGA328P) */
+
+    /* extra return should be unused */
+    return 0;
 }
 
-static void _set_brr(uart_t uart, uint32_t baudrate)
+static inline void _update_brr(uart_t uart)
 {
-    uint16_t brr;
+    dev[uart]->BRR = uart_brr[uart];
 
+#if defined(UART_DOUBLE_SPEED)
+    dev[uart]->CSRA |= (1 << U2X0);
+#endif
+}
+
+static inline void _calc_brr(uart_t uart, uint32_t baudrate)
```

Comment:
is an extra method for this necessary? it's only used a single time

Reply:
It improves readability, and is directly adapted from prior behavior. It should produce the same instructions, because of inline.